### PR TITLE
Refactor Check your answer 'redirect' mechanism

### PR DESCRIPTION
**Pull request by Kalagan**
I had a go at changing the way the "Check your answer" flow works.
By passing a `redirect_path` parameter around rather than using the state of the `legal_aid_application`.
Personally, I think it makes things simpler and we don't have to worry about the state.
Let me know what you think.

## Checklist

Before you ask people to review this PR:

- [x] Tests and rubocop should be passing: `bundle exec rake`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [x] There should be no unneccessary whitespace changes. These make diffs harder to read and conflicts more likely. 
- [x] The PR description should say what you changed and why, with a link to the JIRA story.
- [x] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Comment by reggieb** [comment]
As discussed, my problem with this solution is that if the return is not immediate (for example address modification occurs over two pages - four actions) how do you pass the return information through these steps. This solution uses hidden fields in the affected forms. I think that spreads the logic around too much, in that at the moment only `Steppable` needs to make decisions based on the journey type (normal or check-your-answers), whereas in this solution each form in a multi-page modification, needs to make a decision - each of those forms needs the modification.

In summary, I think this is a perfectly valid way of handling the issue. I just don't think it is better than the current solution.

Having said that, I am hugely biased having written the current solution.

**Comment by stephenrichards** [comment]
I think I agree with Rob that passing hidden fields around spreads the love to widely.

Personally, I'd be in favour of a solution the was system wide, and driven by a yaml config file.  The config file could specify the controller path, give the default Back and Forward paths, and then a series of "predicate methods" which if true, yielded another path.  That way, we have all the details for what goes where in one place, and just one object responsible for determining where to go.

Thoughts?

**Comment by reggieb** [comment]
The main section of `Steppable` is a hash - so could be easily represented via yaml if you want to go that way.

**Comment by Kalagan** [comment]
@stephenrichards ,
Yes I completely agree for the main flow and your predicate methods idea. Not sure if we can define logic in a YAML file though. Right now I'm working on https://dsdmoj.atlassian.net/browse/AP-251
And I'm thinking of building a hash like this:
```
STEPS_FLOW = {
  ...
  other_capital: {
    back: :savings_investments,
    forward: ->(application) { application.own_home_yes? ? :capital_restrictions : :check_answers }
  }
  ...
```

And then another hash to define the paths of each steps:

```
STEP_PATHS: {
  ...
  other_capital: ->(application) { ........_path(application) },
  capital_restrictions: ->(_application) { ........._path }
  ...
}
```

But for the check_your_answer logic, we need something else

**Comment by reggieb** [comment]
Rather than two constants, could we do:
```ruby
STEPS_FLOW = {
  ...
  other_capital: {
    path: ->(application) { ........_path(application) },
    back: :savings_investments,
    forward: ->(application) { application.own_home_yes? ? :capital_restrictions : :check_answers }
  }
```
Also if we're defining procs within the definitions (which I don't have a problem with), then I think we should keep with using hashes rather than moving to using yaml.

**Comment by reggieb** [comment]
Another consideration - is now the time to refactor the citizens journey so that the current legal aid application is passed via the url (as in the providers journey) rather than a session key. Otherwise, we'll have to do a fairly large rejig of the citizens version of steppable later.

**Comment by Kalagan** [comment]
Do we necessarily need to do that?
